### PR TITLE
emit error stats for both client and server sides

### PR DESCRIPTION
**Pull request by zhijinli**
@Raynos @charliezhang

### CODE REVIEW

File: adapters/statsd-report-status-code.js [hunk 1]
```diff
@@ -15,11 +15,13 @@ function handleReportingRequest(request, requestOptions, handleResponse) {
     var self = this;
     self.requestHandler.request(request, requestOptions, onResponse);
     function onResponse(error, response) {
+        var resource = requestOptions.resource;
+        var statsEmitter = self.options.statsEmitter;
         if (error) {
+            // report 500 for client errors, e.g. ETIMEDOUT, ESOCKETTIMEDOUT
+            statsEmitter.emit('statusCode', resource, 500);
```
Comment (zhijinli) [inline adapters/statsd-report-status-code.js]: I was wondering if all the client errors should be considered 500?

Comment (Raynos) [inline adapters/statsd-report-status-code.js]: this is not what we want, we do not want this to be a 500; that hides implementation details.

let's change it to be a new event `'client-error'`.

**Comment by Raynos** [comment]
We also have to emit the same event in the airlock wrapper.

File: adapters/statsd-report-status-code.js (hunk 1 above)
Comment (Raynos) [inline adapters/statsd-report-status-code.js]: https://github.com/uber/typed-request-client/blob/master/adapters/probing.js#L39

Comment (Raynos) [inline adapters/statsd-report-status-code.js]: To help differential between the two we can do `'client-error', 'http'` vs `'client-error', 'airlock'`

Comment (Raynos) [inline adapters/statsd-report-status-code.js]: In fact; for http errors lets just report `err.code` and for airlock lets just report `'airlock'`

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**